src/ui.rs

```diff
@@ -185,99 +181,93 @@ impl epi::App for App {
                 if self.state.token.is_some() && ui.button("Reset").clicked() {
                     self.state.token = None;
                 }
+            });
 
-                match &self._proxy_timer {
-                    Some(timer) if timer.elapsed_milliseconds(1000) => {
-                        self._proxy_timer = None;
-                        self.proxy.send(crate::proxy::Message::Refresh);
+            ui.separator();
+
+            ui.horizontal(|ui| {
+                ui.vertical(|ui| {
+                    if ui
+                        .add(
+                            egui::TextEdit::singleline(&mut self.state.channel)
+                                .hint_text("Channel name"),
+                        )
+                        .changed()
+                    {
+                        self.state.channel.truncate(128);
+                        if !self.state.channel.is_empty() {
+                            self._channel_name_timer = Some(Timer::new());
```

Comment:
I think this should call `update_tts_config` when the user presses `enter` instead of using a `Timer`. There should also be some kind of visual feedback for anything that updates or saves in some way, but that's out of scope for this PR.